src/core/Akka.Tests/Actor/ActorSystemSpec.cs

```diff
@@ -131,10 +141,67 @@ public void Throw_exception_when_register_callback_after_shutdown()
             Assert.Equal("ActorSystem already terminated.", ex.Message);
         }
 
+        [Fact]
+        public void Reliably_create_waves_of_actors()
+        {
+            var timeout = Dilated(TimeSpan.FromSeconds(20));
+            var waves = Task.WhenAll(
+                Sys.ActorOf(Props.Create<Wave>()).Ask<string>(50000),
+                Sys.ActorOf(Props.Create<Wave>()).Ask<string>(50000),
+                Sys.ActorOf(Props.Create<Wave>()).Ask<string>(50000));
+
+            waves.Wait(timeout.Duration() + TimeSpan.FromSeconds(5));
+
+            Assert.Equal(new[] { "done", "done", "done" }, waves.Result);
+        }
+
+        [Fact]
+        public void Find_actors_that_just_have_been_created()
+        {
+            Sys.ActorOf(Props.Create(() => new FastActor(new TestLatch(), TestActor)).WithDispatcher("slow"));
+            Assert.Equal(typeof(LocalActorRef), ExpectMsg<Type>());
+        }
+
+        [Fact]
+        public void Reliable_deny_creation_of_actors_while_shutting_down()
+        {
+            var sys = ActorSystem.Create("DenyCreationWhileShuttingDone");
+            Task.Delay(100).ContinueWith(_ => sys.Terminate());
+            var failing = false;
+            var created = new HashSet<IActorRef>();
+
+            while (!sys.WhenTerminated.IsCompleted)
+            {
+                try
+                {
+                    var t = sys.ActorOf<Terminater>();
+                    Assert.False(failing); // because once failing => always failing (it’s due to shutdown)
+                    created.Add(t);
+
+                    if (created.Count % 1000 == 0)
+                        Thread.Sleep(50); // in case of unfair thread scheduling
+                }
+                catch (InvalidOperationException)
+                {
+                    failing = true;
+                }
+
+                
+                //TODO add check if sys.Uptime >= 5 as far as this property is implemented
+                //if (!failing)
+                //    throw new AssertionFailedException(created.Last() + Environment.NewLine +
+                //                                       "System didn't terminate within 5 seconds");
+            }
+
+            Assert.Empty(
+                created.Cast<ActorRefWithCell>()
+                    .Where(actor => !actor.IsTerminated && !(actor.Underlying is UnstartedCell)));
```

Comment:
During this test ~1500 actors are created, from which ~1300 are terminated but the remaining ~200 haven't a ```UnstartedCell``` but rather a normal ```ActorCell```